src/controllers/tasks.controller.js

```diff
@@ -0,0 +1,183 @@
+/* eslint-disbled */
+import TaskModel from '../models/task.model.js';
+import faker from 'faker';
+
+export function findAllTasks() {
+  console.log('CONTROLLER: trying to get all the tasks ');
+  return TaskModel.find();
+}
+
+export function findTask(req, res) {
+  console.log(' CONTROLLER: get a task ');
+  const { id } = req.params;
+
+  TaskModel.findById(id)
+    .then((data) => {
+      console.log(data);
+      res.status(200).send(data);
+    })
+    .catch((err) => {
+      res.status(404).send({ message: err.message || 'task not found' });
+    });
+}
```

Comment:
I wouldn't pass `req` and `res` objects to the `controller`. Just pass the parameter you want to use, and let the `controller` handle the error (throwing an error probably). Similarly, handle the response here

```suggestion
export function findTask(id) {
  logger.info(' CONTROLLER: get a task'); // logger library
  return TaskModel.findById(id)
    .then((data) => data)
    .catch((err) => {
      logger.error(`Task with id ${id} not found`);
      return new Error ('task not found');
    });
}
```